pkg/storegateway/indexcache/cache.go

```diff
@@ -91,23 +92,22 @@ type PostingsKey string
 
 // CanonicalPostingsKey creates a canonical version of PostingsKey
 func CanonicalPostingsKey(postings []storage.SeriesRef) PostingsKey {
-	hashable := make([]byte, len(postings)*8)
-	for i, posting := range postings {
-		for octet := 0; octet < 8; octet++ {
-			hashable[i+octet] = byte(posting >> (octet * 8))
-		}
-	}
-
-	// We hash the postings list twice to minimize the chance of collisions
-	hasher1, _ := blake2b.New256(nil) // This will never return an error
-	hasher2 := sha1.New()
-
-	_, _ = hasher1.Write(hashable)
-	_, _ = hasher2.Write(hashable)
-
-	checksum := hasher2.Sum(hasher1.Sum(nil))
+	hashable := unsafeCastPostingsToBytes(postings)
+	checksum := blake2b.Sum256(hashable)
+	return PostingsKey(base64.RawURLEncoding.EncodeToString(checksum[:]))
+}
 
-	return PostingsKey(base64.RawURLEncoding.EncodeToString(checksum))
+const bytesPerPosting = int(unsafe.Sizeof(storage.SeriesRef(0)))
+
+// unsafeCastPostingsToBytes returns the postings as a slice of bytes with minimal allocations.
+// It casts the memory region of the underlying array to a slice of bytes.
```

Comment:
nit: Can you expand this a bit to be explicit about requirements for the caller?
```suggestion
// It casts the memory region of the underlying array to a slice of bytes. The resulting byte slice is only valid as long as the postings slice exists and is unmodified.
```